### DISCUSSION TITLE
MacOS version

### DISCUSSION THREAD

**Discussion by rohitsaha29** [Alex313031/Mercury#24] [General]
will there be a MacOS version possibly?

**Comment by gz83** [comment]
Whether to provide a MacOS version for mercury is up to Alex and midzer

@Alex313031 @midzer

**Reply by j03ll0b0** [reply]
please help us get mercurial updated to build under mac os, I get this:

0:55.20 DEBUG: <truncated - see config.log for full output>
 0:55.20 DEBUG: | pub extern fn hello() { println!("Hello world"); }
 0:55.20 DEBUG: Executing: `/Users/usuario/.rustup/toolchains/stable-x86_64-apple-darwin/bin/rustc --crate-type staticlib --target=x86_64-unknown-linux-gnu -o /var/folders/6x/9r8h31t57lj1y32l0t34cb7c0000gn/T/conftest6q9a2j2q.rlib /var/folders/6x/9r8h31t57lj1y32l0t34cb7c0000gn/T/conftestqhddi_l3.rs`
 0:55.21 DEBUG: The command returned non-zero exit status 1.
 0:55.21 DEBUG: Its error output was:
 0:55.21 DEBUG: | error[E0463]: can't find crate for `std`
 0:55.21 DEBUG: |   |
 0:55.21 DEBUG: |   = note: the `x86_64-unknown-linux-gnu` target may not be installed
 0:55.21 DEBUG: |   = help: consider downloading the target with `rustup target add x86_64-unknown-linux-gnu`
 0:55.21 DEBUG: |
 0:55.21 DEBUG: | error: cannot find macro `println` in this scope
 0:55.21 DEBUG: |  --> /var/folders/6x/9r8h31t57lj1y32l0t34cb7c0000gn/T/conftestqhddi_l3.rs:1:25
 0:55.24 DEBUG: |   |
 0:55.25 DEBUG: | 1 | pub extern fn hello() { println!("Hello world"); }
 0:55.25 DEBUG: |   |                         ^^^^^^^
 0:55.25 DEBUG: |
 0:55.25 DEBUG: | error: requires `sized` lang_item
 0:55.25 DEBUG: |
 0:55.25 DEBUG: | error: aborting due to 3 previous errors
 0:55.25 DEBUG: |
 0:55.26 DEBUG: | For more information about this error, try `rustc --explain E0463`.
 0:55.26 ERROR: Cannot compile for x86_64-pc-linux-gnu with /Users/usuario/.rustup/toolchains/stable-x86_64-apple-darwin/bin/rustc
 0:55.26 The target may be unsupported, or you may not have
 0:55.26 a rust std library for that target installed. Try:
 0:55.26   rustup target add x86_64-unknown-linux-gnu
*** Fix above errors and then restart with "./mach build"
 0:55.60 W Exception when writing resource usage file: [Errno 2] No such file or directory: '/Users/usuario/mozilla-unified/obj-x86_64-pc-linux-gnu/instrumented/.mozbuild/profile_build_resources.json'
 ✘ usuario@EBG5  ~/mozilla-unified  ☿ 764037@default ± 

**Reply by gz83** [reply]
We currently do not have suitable compilation parameters customized for MacOS systems

@j03ll0b0

**Comment by j03ll0b0** [comment]
perhaps I'll have to get used to run it with docker + x11

**Reply by peterjumper** [reply]
it this as fast as thorium in mac?
tried but also failed, completely no clue about installing this , but i think this is still legitimate though
